### PR TITLE
Update Travis configuration

### PR DESCRIPTION
**Pull request by Geod24**
This does a couple of cleanup to the travis configuration.
See the linked blog posts for the non-D-specific parts.

### CODE REVIEW

**Comment by Geod24** [comment]
Hum, there's a lot of duplication in the readme as to what's supported.
Shouldn't we state the goal and point to the travis file instead ?

**Comment by Geod24** [comment]
Well I give up on GDC for the time being...

File: .travis.yml [hunk 1]
```diff
@@ -34,15 +34,14 @@ d:
   - ldc-1.9.0
   - ldc-1.8.0
   - ldc-1.7.0
-  - ldc-1.6.0
+  - dmd-2.085
```
Comment (s-ludwig) [inline .travis.yml]:
```suggestion
  - dmd-2.085.1
```

**Comment by s-ludwig** [comment]
There is also appveyor.yml with an additional set of duplicate entries, but since updates are relatively rare, it didn't really bother me so far.

**Comment by s-ludwig** [comment]
Looks like the DMD 2.085.1/vibe-core failure is genuine, unfortunately, I'd suggest to simply allow it to fail for now:
```
matrix:
    allow_failures:
        - d: 2.085.1
          env: VIBED_DRIVER=vibe-core
```
(hopefully got the syntax right there)

**Comment by s-ludwig** [comment]
Looks like the DMD 2.085.1/vibe-core failure is genuine, unfortunately, I'd suggest to simply allow it to fail for now:
```
matrix:
    allow_failures:
        - d: 2.085.1
          env: VIBED_DRIVER=vibe-core
```
(hopefully got the syntax right there)

**Comment by Geod24** [comment]
@s-ludwig : That didn't work. Did a bit of digging and all parts of the build need to be matched, so I added `PARTS` too but it didn't work either. Why don't we just stick to v2.085.0 ? v2.085.1 is not going to be fixed anyway.

**Comment by s-ludwig** [comment]
Okay, not worth the trouble then, lets stick with 2.085.0.

**Comment by s-ludwig** [comment]
Finally green 🎉 That looked like a lot more effort than it should be, thanks for finishing this!

**Pull request merged**